docs/DeveloperGuide.md

```diff
@@ -1,38 +1,190 @@
+    - Saves all the user data into `user.json` file that is saved in the storage of user's computer when the user inputs the "bye" command.
+
+3. **Behavior**: The `DataStorage` class acts as an intermediary between the application and the stored lists of components and PC builds. It helps to maintain a clean separation of concerns by providing a unified interface for handling data storage and retrieval. This allows the rest of the application to interact with the data easily.
+
+In summary, the `DataStorage` class is responsible for managing lists of components and PC builds, providing methods to load all the data needed, and interact with this loaded data.
+
+#### loadChassis() method
+This Sequential analysis will show how the loadChassis() method works, this will also serve as an example for all the other load methods that the DataStorage class has.
+
+The `loadChassis` method reads chassis information from a JSON file and populates a `ChassisList` with the data.
+
+When the `loadChassis` method is called with a file path as its argument, it first gets a `ClassLoader` instance from the `DataStorage` class and uses it to create an `InputStream` for the specified file. A `BufferedReader` is then created to read the contents of the file.
+
+The JSON file is read and concatenated into a single string using `reduce`. The GSON library is then used to deserialize the JSON string into a `Map<String, Chassis>`, with the key being the name of the chassis and the value being a `Chassis` object.
+
+Next, the method iterates over the `chassisMap` and sets the name attribute of each `Chassis` object to its corresponding key in the map.
+
+Finally, the method iterates over the `chassisMap` again, adding each `Chassis` object to the `ChassisList` stored in the `stringToComponentListMap` HashMap.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+
+![load Chassis Sequential Diagram](uml-pictures/loadChassisSequential.png)
+
+### Parser class
+The UML class diagram shows the features of the Parser class:
+
+![Parser Class Diagram](uml-pictures/ParserClass.png)
+
+The purpose of the Parser class is to parse the user input into commands for execution. With reference to the diagram,
+the Parser class does not have any attributes and only contains public methods for parsing of the user input.
+The methods in the Parser class creates a subclass which inherits from the abstract superclass Command class. 
+From the diagram, Command class has a private attribute named arguments which stores the user inputs as a String. The Command class
+contains methods that help to set and retrieve the private attribute in the class. 
+Overall, this class diagram provides an overview of the purpose of the Parser class.
+
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
```

Comment:
When viewing this Class diagram on the webpage, it is difficult to read all the components clearly without zooming in.
Maybe, the diagrams could be split into multiple and presented in a different way to make it more readable.

Reply:
TA said this was ok